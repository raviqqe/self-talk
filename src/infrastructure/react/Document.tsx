import { MdEdit } from "react-icons/md";
import React, { useState } from "react";
import styled from "styled-components";
import { IconButton } from "./IconButton";
import { Markdown } from "./Markdown";
import {
  UpdateDocument,
  IProps as IUpdateDocumentProps,
} from "./UpdateDocument";
import { boxShadow } from "./style";
import { white } from "./style/colors";

const Container = styled.div`
  ${boxShadow};
  background: ${white};
  padding: 1em;
  padding-right: 2.1em;
  border-radius: 0.5em;
  position: relative;
`;

const ButtonContainer = styled.div`
  display: flex;
  position: absolute;
  top: 0.4em;
  right: 0.4em;
`;

interface IProps extends IUpdateDocumentProps {}

export const Document = ({
  document,
  insertFiles,
  updateDocument,
  ...restProps
}: IProps) => {
  const [editing, setEditing] = useState(false);

  return editing ? (
    <UpdateDocument
      document={document}
      insertFiles={insertFiles}
      updateDocument={async (document) => {
        setEditing(false);
        await updateDocument(document);
      }}
      {...restProps}
    />
  ) : (
    <Container {...restProps}>
      <Markdown>{document.text}</Markdown>
      <ButtonContainer>
        <IconButton aria-label="Edit" onClick={() => setEditing(true)}>
          <MdEdit />
        </IconButton>
      </ButtonContainer>
    </Container>
  );
};
