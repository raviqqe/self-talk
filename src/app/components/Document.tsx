import { styled } from "@linaria/react";
import { type JSX, useState } from "react";
import { MdEdit } from "react-icons/md";
import type * as domain from "../../domain.js";
import { documentUpdater } from "../../main/document-updater.js";
import { boxShadow, white } from "../style.js";
import { IconButton } from "./IconButton.js";
import { Markdown } from "./Markdown.js";
import { UpdateDocument } from "./UpdateDocument.js";

const Container = styled.div`
  ${boxShadow};
  background: ${white};
  padding: 1em;
  padding-right: 2.1em;
  border-radius: 0.5em;
  position: relative;
`;

const ButtonContainer = styled.div`
  display: flex;
  position: absolute;
  top: 0.4em;
  right: 0.4em;
`;

interface Props {
  className?: string;
  document: domain.Document;
}

export const Document = ({ document, ...rest }: Props): JSX.Element => {
  const [editing, setEditing] = useState(false);

  return editing ? (
    <UpdateDocument
      document={document}
      onSubmit={async (document) => {
        setEditing(false);
        await documentUpdater.update(document);
      }}
      {...rest}
    />
  ) : (
    <Container {...rest}>
      <Markdown>{document.text}</Markdown>
      <ButtonContainer>
        <IconButton aria-label="Edit" onClick={() => setEditing(true)}>
          <MdEdit />
        </IconButton>
      </ButtonContainer>
    </Container>
  );
};
