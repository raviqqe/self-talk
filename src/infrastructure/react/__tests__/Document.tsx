import { create } from "react-test-renderer";
import { Document } from "../Document";

it("renders", () => {
  expect(
    create(
      <Document
        document={{ id: "id", text: "text" }}
        insertFiles={async () => "url"}
        updateDocument={async () => {}}
      />
    ).toJSON()
  ).toMatchSnapshot();
});
