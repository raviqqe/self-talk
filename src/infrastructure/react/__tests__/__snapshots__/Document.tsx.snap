// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="sc-AxmLO ehOlxA"
>
  <div
    className="sc-AxirZ hvfzc"
  >
    <p>
      text
    </p>
  </div>
  <div
    className="sc-fzozJi eyZSoi"
  >
    <div
      aria-label="Edit"
      className="sc-AxjAm ijbaQh"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
        />
      </svg>
    </div>
  </div>
</div>
`;
